src/registrar/models/dns/dns_zone_vendor_dns_zone.py

```diff
@@ -1,7 +1,13 @@
-from django.db.models import ForeignKey, CASCADE
+from django.db.models import ForeignKey, BooleanField, CASCADE, UniqueConstraint, Q
 from ..utility.time_stamped_model import TimeStampedModel
 
 
 class DnsZone_VendorDnsZone(TimeStampedModel):
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=["dns_zone"], condition=Q(is_active=True), name="unique_active_vendor_per_dns_zone")
```

Comment:
```suggestion
            UniqueConstraint(fields=["dns_zone"], condition=Q(is_active=True), name="unique_active_vendor_zone_per_dns_zone")
```
(I know I had what you have here in the ticket- realized this is more accurate)